ui/app/controllers/jobs/job/definition.js

```diff
@@ -1,19 +1,30 @@
 import Controller from '@ember/controller';
-import WithNamespaceResetting from 'nomad-ui/mixins/with-namespace-resetting';
+import { action } from '@ember/object';
 import { alias } from '@ember/object/computed';
-import classic from 'ember-classic-decorator';
 import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import classic from 'ember-classic-decorator';
+import WithNamespaceResetting from 'nomad-ui/mixins/with-namespace-resetting';
 
 @classic
 export default class DefinitionController extends Controller.extend(
   WithNamespaceResetting
 ) {
+  @tracked view = this.specification ? 'job-spec' : 'full-definition';
+  queryParams = ['view'];
+
   @alias('model.definition') definition;
   @alias('model.job') job;
   @alias('model.specification') specification;
 
   @service router;
 
+  @action
+  toggleView() {
+    const opposite = this.view === 'job-spec' ? 'full-definition' : 'job-spec';
+    this.view = opposite;
+  }
```

Comment:
Undone in https://github.com/hashicorp/nomad/pull/16711

Reply:
This branch was solely responsible for setting up the query parameter interface. #16711 is responsible for converting the `toggle` logic into `select` logic. I'm practicing atomic commits.